.devcontainer/devcontainer.json

```diff
@@ -0,0 +1,29 @@
+// For format details, see https://aka.ms/devcontainer.json. For config options, see the
+// README at: https://github.com/devcontainers/templates/tree/main/src/python
+{
+	"name": "Python 3",
+	"image": "mcr.microsoft.com/devcontainers/python:0-3.10",
+	"features": {
+		"ghcr.io/devcontainers/features/node:1": {}
+	},
+
+	// Features to add to the dev container. More info: https://containers.dev/features.
+	// "features": {},
+
+	// Use 'forwardPorts' to make a list of ports inside the container available locally.
+	"forwardPorts": [8000],
+
+	// Use 'postCreateCommand' to run commands after the container is created.
+	"postCreateCommand": "pip3 install --user -r requirements.txt && npm install && mkdir -p data && python3 manage.py migrate",
```

Comment:
linkding also has a few integration tests using Playwright. Playwright needs to additionally download a browser, currently I'm using `playwright install chromium` in Github Actions. Doing that in the dev container works, but then the tests fail:
```
python manage.py test

...

╔══════════════════════════════════════════════════════╗
║ Host system is missing dependencies to run browsers. ║
║ Missing libraries:                                   ║
║     libatk-1.0.so.0                                  ║
║     libatk-bridge-2.0.so.0                           ║
║     libcups.so.2                                     ║
║     libdbus-1.so.3                                   ║
║     libatspi.so.0                                    ║
║     libXcomposite.so.1                               ║
║     libXdamage.so.1                                  ║
║     libXfixes.so.3                                   ║
║     libXrandr.so.2                                   ║
║     libgbm.so.1                                      ║
║     libdrm.so.2                                      ║
║     libxkbcommon.so.0                                ║
║     libpango-1.0.so.0                                ║
║     libcairo.so.2                                    ║
║     libasound.so.2                                   ║
╚══════════════════════════════════════════════════════╝
```

Not a blocker, the integration test setup currently also has other issues like needing to run certain commands beforehand. But maybe something to look into in the future.

Reply:
I have added "playwright install chromium && playwright install-deps" to the post create script. The tests run now, however there were 2 failures and 1 error. I don't know whether those are existing problems or whether they are to do with running in the dev container.

If there are other commands that need to be run prior to the tests let me know and I can try adding them in. If post create isn't appropriate there are also some other hooks where scripts can be run.